Arm/Syntax.lean

```diff
@@ -0,0 +1,16 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Siddharth Bhat
+
+Provide convenient syntax for writing down state manipulation in ARM programs.
+-/
+import Arm.State
+
+namespace ArmStateNotation
+
+/-! We build a notation for `read_mem_bytes $n $base $s` as `$s[$base, $n]` -/
```

Comment:
Question: does this notation also show up in the goals? Or do we still see `read_mem_bytes...` there?

Reply:
I don't think we'll see it in the goals, since I wrote it as `syntax` plus `macro_rules` instead of `notation`, precisely because I wasn't sure if people want to see it in the goal state.